cartofacade/cgo_api.go

```diff
@@ -0,0 +1,456 @@
+// Package cartofacade provides an api to call into c code
+//
+//nolint:lll
+package cartofacade
+
+/*
+	#cgo CFLAGS: -I../viam-cartographer/src/carto_facade
+
+	// the libraries that need to be linked can be dermined by checking line 258 of the build.ninja file that is autogenerated during make build
+  	#cgo LDFLAGS: -L../viam-cartographer/build -L../viam-cartographer/cartographer/build -lcartographer  -lviam-cartographer  -labsl_raw_hash_set -labsl_hash  -labsl_str_format_internal  -labsl_synchronization    -labsl_strings  -labsl_throw_delegate -lcairo -llua -lstdc++ -lceres -lprotobuf -lglog -lboost_filesystem -lboost_iostreams -lpcl_io -lpcl_common
+
+	#include "../viam-cartographer/src/carto_facade/carto_facade.h"
+
+	viam_carto_lib* alloc_viam_carto_lib() { return (viam_carto_lib*) malloc(sizeof(viam_carto_lib)); }
+	void free_alloc_viam_carto_lib(viam_carto_lib* p) { free(p); }
+
+	viam_carto* alloc_viam_carto() { return (viam_carto*) malloc(sizeof(viam_carto)); }
+	void free_alloc_viam_carto(viam_carto* p) { free(p); }
+
+	bstring* alloc_bstring_array(size_t len) { return (bstring*) malloc(len * sizeof(bstring)); }
+	void free_bstring_array(bstring* p, int len) {
+		for (int i = 0; i < len; i++) {
+			bdestroy(p[i]);
+		}
+		free(p);
+	}
+*/
+import "C"
+
+import (
+	"errors"
+	"time"
+	"unsafe"
+)
+
+// CartoLib holds the c type viam_carto_lib
+type CartoLib struct {
+	value *C.viam_carto_lib
+}
+
+// Carto holds the c type viam_carto
+type Carto struct {
+	value *C.viam_carto
+}
+
+// getPointCloudMapResponse is a struct to hold the c type viam_carto_get_point_cloud_map_response
+type getPointCloudMapResponse struct {
```

Comment:
Can we please move this to the private types?

Reply:
actually, lets remove the private types which just wrap go types

Reply:
removing this type all together